anomalib/models/dfkde/model.py

```diff
@@ -36,17 +94,14 @@ class DfkdeLightning(AnomalyModule):
 
     def __init__(self, hparams: Union[DictConfig, ListConfig]):
         super().__init__(hparams)
-        self.threshold_steepness = 0.05
-        self.threshold_offset = 12
+        threshold_steepness = 0.05
+        threshold_offset = 12
 
-        self.backbone = getattr(torchvision.models, hparams.model.backbone)
-        self.feature_extractor = FeatureExtractor(backbone=self.backbone(pretrained=True), layers=["avgpool"]).eval()
-
-        self.normality_model = NormalityModel(
-            filter_count=hparams.model.max_training_points,
-            threshold_steepness=self.threshold_steepness,
-            threshold_offset=self.threshold_offset,
+        backbone = getattr(torchvision.models, hparams.model.backbone)
```

Comment:
Maybe you could create the backbone inside `DfkdeModel` and just pass the `hparams.model.backbone` to the constructor of that class. That would be more in line with other models (see Padim for example).